### PR TITLE
[Torch] fix retrieve error message segfault

### PR DESCRIPTION
**Pull request by fortianyou**
We met segfault when retrieving error message after build engine failed. To make our process more robust we would like to enable log message only when TORCH_BLADE_DEBUG_LOG==on

### CODE REVIEW

File: pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp [hunk 1]
```diff
@@ -132,7 +132,15 @@ TrtUniquePtr<nvinfer1::ICudaEngine> TensorrtOnnxParser::BuildEngine(
...
-    if (trt_engine == nullptr) {
+
+    // We had meet segfault when retrive error message after build engine
+    // failed. To make our process more robust we would like to enable log
+    // message only when TORCH_BLADE_DEBUG_LOG==on
+    const char* debug_log_flag = std::getenv("TORCH_BLADE_DEBUG_LOG");
+    bool enable_debug_log = debug_log_flag != nullptr &&
+        (std::strcmp(debug_log_flag, "on") == 0 ||
+         std::strcmp(debug_log_flag, "true") == 0);
+    if (trt_engine == nullptr && enable_debug_log) {
```
Comment (linearhit) [inline pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp:143]: nit: if it's tricky to use tensorflow::ReadBoolFromEnvVar()
We'd better make it into some util files.

File: pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp [hunk 2]
```diff
@@ -132,7 +132,15 @@ TrtUniquePtr<nvinfer1::ICudaEngine> TensorrtOnnxParser::BuildEngine(
...
         context.builder->buildEngineWithConfig(*context.network, *config));
-    if (trt_engine == nullptr) {
+
+    // We had meet segfault when retrive error message after build engine
+    // failed. To make our process more robust we would like to enable log
+    // message only when TORCH_BLADE_DEBUG_LOG==on
+    const char* debug_log_flag = std::getenv("TORCH_BLADE_DEBUG_LOG");
+    bool enable_debug_log = debug_log_flag != nullptr &&
+        (std::strcmp(debug_log_flag, "on") == 0 ||
+         std::strcmp(debug_log_flag, "true") == 0);
```
Comment (Orion34-lanbo) [inline pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp:142]: Nit: perhaps add the following condition as well
```
std::strcmp(debug_log_flag, "1") == 0
```

File: pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp (hunk 1 above)
Comment (fortianyou) [reply pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp:143]: Done.

File: pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp [hunk 3]
```diff
@@ -134,13 +134,20 @@ TrtUniquePtr<nvinfer1::ICudaEngine> TensorrtOnnxParser::BuildEngine(
     }
     auto trt_engine = TrtUniquePtr<nvinfer1::ICudaEngine>(
         context.builder->buildEngineWithConfig(*context.network, *config));
-    if (trt_engine == nullptr) {
-      LOG(INFO) << "Failed to build the engine, error message are:";
+
+    // We had meet segfault when retrive error message after build engine
+    // failed. It's a issue of TensorRT.
+    // To make our process more robust we would like to enable log
```
Comment (bddppq) [inline pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp:140]: When `TORCH_BLADE_DEBUG_LOG` is set, the following code will still try to get error msg and run into the same segfault?

Comment (fortianyou) [reply pytorch_blade/src/compiler/tensorrt/bridge/tensorrt_onnx_parser.cpp:140]: Yes, but occasionally. We had only run into the segfault when trying to optimize TransformerXL.

**Review by linearhit — APPROVED** [review]
LGTM

**Pull request merged**